interpreter/src/lib.rs

```diff
@@ -37,44 +39,117 @@ impl State {
         self.memory.insert(x.to_string(), value);
     }
 
-    pub fn write(&mut self, x: &str, value: u32) {
+    pub fn write(&mut self, x: &str, value: u32, thread: &str) {
+
         match self.memory_model {
             MemoryModel::Sc => {
                 self.memory.insert(x.to_string(), value);
             }
             MemoryModel::Tso => {
-                todo!()
+                if let Some(buffer) = self.write_buffers.get_mut(thread) {
+                    buffer.insert(x.to_string(), value);
+                } else {
+                    self.memory.insert(x.to_string(), value);
+                }
             }
         }
     }
 
     pub fn write_local(&mut self, thread: &str, x: &str, value: u32) {
-        self.memory.insert(format!("{thread}.{x}"), value);
+        match self.memory_model {
+            MemoryModel::Sc => {
+                self.memory.insert(format!("{thread}.{x}"), value);
+            }
+            MemoryModel::Tso => {
+                if let Some(buffer) = self.write_buffers.get_mut(thread) {
+                    buffer.insert(format!("{thread}.{x}"), value);
+                } else {
+                    //
+                }
+            }
+        }
```

Comment:
I don't think local variables need to interact with the write buffers since they mainly act as intermediate values and are not present in the aeg